### DISCUSSION TITLE
How does lichess verify the authenticity of analysis?

### DISCUSSION THREAD

**Discussion by manoharprabhu** [lichess-org/fishnet#233] [Q&A] [answered]
How does lichess trust the analysis result received from the fishnet clients?

What happens if a position is not evaluated correctly either due to hardware fault or malicious manipulation before sending it over the network?

**Comment by niklasf** [comment, marked as answer]
All clients are trusted by default. With no leaderboards and no way to target particular games, there is very little incentive for malicious manipulation. Keys are only given to established Lichess accounts.

Since Stockfish is used in such a way that the results are deterministic, it's possible to run spot checks if required, for example if a user reports incorrect analysis.

**Comment by John-Set-Dough** [comment]
Yes, all clients are trusted by default. Still users get their unique private key that are bound to the user. So malicious users can be blocked I guess.
